src/Common/src/TypeSystem/Common/TypeSystemConstraintsHelpers.cs

```diff
@@ -97,6 +103,32 @@ private static bool CheckGenericSpecialConstraint(TypeDesc type, GenericConstrai
             return false;
         }
 
+        private static List<TypeDesc> GetInstantiatedConstraints(TypeDesc type, Instantiation typeInstantiation, Instantiation methodInstantiation)
+        {
+            var instantiatedConstraints = new List<TypeDesc>();
```

Comment:
We tend to use our own `ArrayBuilder<TypeDesc>` in the TypeSystem - it does not allocate anything if the result is empty.